### DISCUSSION TITLE
How to run unreal editor without a gpu

### DISCUSSION THREAD

**Discussion by seisdr** [adamrehn/ue4-docker#352] [Q&A] [answered]
I'm trying to run unreal engine on a dedicated server that doesn't contain a GPU
I downloaded it from https://www.unrealengine.com/en-US/linux
Vulkan failed to select physical after passing profile checks 

Ubuntu 22.04.3
Device: llvmpipe (LLVM 15.0.7, 256 bits) (0xffffffff)                                                                               Version: 23.2.1

I'm sorry if this is not relevant but i couldn't find an answer,

**Comment by seisdr** [comment]
I also tried ue4-docker to build unreal engine but that failed 
```
--------------------                                                                                      ERROR: failed to solve: process "/bin/sh -c CHANGELIST=\"$CHANGELIST\" \tmkdir /home/ue4/UnrealEngine && \tcd /home/ue4/UnrealEngine && \tgit init && \tgit remote add origin \"$GIT_REPO\" && \tgit fetch --progress --depth 1 origin \"$GIT_BRANCH\" && \tgit checkout FETCH_HEAD && echo '' && echo 'RUN directive complete. Docker will now commit the filesystem layer to disk.' && echo 'Note that for large filesystem layers this can take quite some time.' && echo 'Performing filesystem layer commit...' && echo ''" did not complete successfully: exit code: 128
[ue4-docker build] Error: failed to build image "adamrehn/ue4-source:4.25.0-opengl-ubuntu22.04".
```

**Reply by adamrehn** [reply]
If you are having issues building images with ue4-docker then please open a separate issue to discuss that problem.

**Comment by adamrehn** [comment, marked as answer]
This is not particularly relevant to ue4-docker since it's a general Unreal Engine question, but I figure this information might still be useful for a number of ue4-docker users.

The Unreal Engine's Vulkan RHI will refuse to allow software rendering by default, resulting in a crash when running in an environment without a GPU. You can specify the command-line flag `-AllowCPUDevices` to permit software rendering (see the relevant logic here: https://github.com/EpicGames/UnrealEngine/blob/5.3.2-release/Engine/Source/Runtime/VulkanRHI/Private/VulkanRHI.cpp#L233-L243), although there is no guarantee that the Editor will run correctly since Epic Games do not test this configuration. If it does work, performance will be significantly worse than when rendering with a GPU.
